src/components/JfListMakerComponent.spec.js

```diff
@@ -1,50 +1,389 @@
-import {shallowMount} from '@vue/test-utils';
+import {shallowMount, createLocalVue} from '@vue/test-utils';
 import JfListMakerComponent from './JfListMakerComponent.vue';
 
+/* Stubs custom directive */
+const localVue = createLocalVue();
+localVue.directive('jf-tooltip-on-overflow', {
+    bind: function() {}
+});
+
+// Utility method to avoid code duplication
+const mountComponent = function( props ){
+    let options = {
+        localVue,
+        propsData: Object.assign({}, props),
+        stubs: ['jf-field','jf-ui-select']
+    }
+    return shallowMount(JfListMakerComponent, options);
+}
+
+//selectors
+const SELECTOR_A_LIST_ROW = `.list-maker-row-value`;
+const SELECTOR_ERROR_MESSAGE = `.jf-validation`;
+//Buttons
+const BUTTON_ADD = '.icon.icon-new';
+const BUTTON_DELETE = '.icon-close';
+//Inputs
+const INPUT_TEXT = `input.input-text`;
+const INPUT_DROPDOWN = 'jf-ui-select-stub'//This is -stub since the child is stubbed
+
+//Events
+const EVENT_ON_ADD_VALUE = 'on-add-value';
+const EVENT_ON_AFTER_ADD_VALUE = 'on-after-add-value';
+const EVENT_INPUT = 'input';
+const EVENT_ON_AFTER_DELETE_VALUE = 'on-after-delete-value';
+
+//Messages
+const MSG_MUST_INPUT_VALUE = `Must input value`;
+const MSG_VALUE_ALREADY_EXISTS = `Value already exists`
+
 describe('JfListMakerComponent', () => {
-    it('renders a form', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs: ['jf-field']
+    describe("display", function() {
+        const wrapper = mountComponent();
```

Comment:
You have to wrap the setup in beforeEach as currently is you setup once and before each test, your tests are not clean and effecting each other.

Reply:
ah! you're right. will do.